pinot-core/src/main/java/org/apache/pinot/core/segment/processing/mapper/SegmentMapper.java

```diff
@@ -71,21 +71,28 @@ public class SegmentMapper {
   private final boolean _includeNullFields;
   private final int _numSortFields;
   private final RecordEnricherPipeline _recordEnricherPipeline;
-  private final CompositeTransformer _recordTransformer;
-  private final ComplexTypeTransformer _complexTypeTransformer;
+  private final TransformPipeline _transformPipeline;
   private final TimeHandler _timeHandler;
   private final Partitioner[] _partitioners;
   private final String[] _partitionsBuffer;
   // NOTE: Use TreeMap so that the order is deterministic
   private final Map<String, GenericRowFileManager> _partitionToFileManagerMap = new TreeMap<>();
-  private AdaptiveSizeBasedWriter _adaptiveSizeBasedWriter;
-  private List<RecordReaderFileConfig> _recordReaderFileConfigs;
-  private List<RecordTransformer> _customRecordTransformers;
+  private final AdaptiveSizeBasedWriter _adaptiveSizeBasedWriter;
+  private final List<RecordReaderFileConfig> _recordReaderFileConfigs;
 
   public SegmentMapper(List<RecordReaderFileConfig> recordReaderFileConfigs,
       List<RecordTransformer> customRecordTransformers, SegmentProcessorConfig processorConfig, File mapperOutputDir) {
+    this(recordReaderFileConfigs,
+        new TransformPipeline(
```

Comment:
Are we clear on how users will pass custom transformers? Up until now our defaults were covering everything which hid this from users of SegmentProcessorFramework.  If we know for sure what defaults should be applied, we should pass that here instead of having every SPF user to pass that. 

No-Op/empty list is one usecase but other than that Im not sure how users will know what exactly to use.

Reply:
We anyways have the simplified constructor which takes care of passing all the default values as before so flows that don't want to tweak the default behaviour can still use that. For flows that require some customisations like for example let's say process segment data without applying the filters (`FilterTransformer`), we need a way to pass a customised `transformPipeline` as well hence the new constructor.
Right now we have `TransformPipeline.getPassThroughPipeline()` for the No-OP usecase you mentioned

Reply:
That's a good point, will extend this so that anyone can pass the required pipeline to `SegmentProcessorFramework` itself.